packages/account/src/Sections/Profile/PersonalDetails/__tests__/personal-details.spec.tsx

```diff
@@ -0,0 +1,220 @@
+import React from 'react';
+import { cleanup, render, waitFor, screen, fireEvent } from '@testing-library/react';
+import { createBrowserHistory } from 'history';
+import { Router } from 'react-router';
+import { PersonalDetailsForm } from '../personal-details-form';
+import { StoreProvider, mockStore } from '@deriv/stores';
+import userEvent from '@testing-library/user-event';
+
+afterAll(cleanup);
+jest.mock('@deriv/components', () => ({
+    ...jest.requireActual('@deriv/components'),
+    Loading: () => <div>Loading</div>,
+}));
+
+jest.mock('@deriv/shared/src/services/ws-methods', () => ({
+    __esModule: true, // this property makes it work,
+    default: 'mockedDefaultExport',
+    WS: {
+        wait: (...payload: []) => Promise.resolve([...payload]),
+    },
+    useWS: () => undefined,
+}));
+
+describe('<PersonalDetailsForm />', () => {
+    const history = createBrowserHistory();
+
+    const promise = Promise.resolve();
+    const fetchResidenceList = jest.fn(() => promise);
+    const fetchStatesList = jest.fn(() => promise);
```

Comment:
**NIT**: May be u can use `jest.fn().mockResolvedValue` or `jest.fn().mockResolvedValueOnce` to return a resolved Promise

Reply:
I think last time Evgeniy requested to change this but it throws error. have to use Promise.resolve() here